feature/s3/manager/download.go

```diff
@@ -166,6 +173,13 @@ func (d Downloader) Download(ctx context.Context, w io.WriterAt, input *s3.GetOb
 		option(&impl.cfg)
 	}
 
+	// Ensures we don't need nil checks later on
+	if impl.cfg.Logger == nil {
+		impl.cfg.Logger = logging.Noop{}
+	} else {
+		impl.cfg.Logger = logging.WithContext(ctx, impl.cfg.Logger)
+	}
```

Comment:
Can this use the generic `logging.WithContext`?  As in if the logger passed in is nil, return a no-op logger? Would prevent duplicating this block between s3 manager and signer.

Reply:
I personally find that a little strange as WithContext is meant to either assert the context interface and give the opportunity to make the logger context aware or return the original logger unmodified. I feel like this overloads the concerns for this function.